eng/jobs/finalize-publish.yml

```diff
@@ -7,6 +7,7 @@ jobs:
     condition: and(
                   succeeded('Build_Linux_Arm'),
```

Comment:
Do these conditions need to be specified at all?  From https://docs.microsoft.com/en-us/azure/devops/pipelines/process/multiple-phases?view=azure-devops&tabs=yaml#conditions

> By default, a job runs if it does not depend on any other job, or if all of the jobs that it depends on have completed and succeeded.

Reply:
Heh, I actually just put in a change in my mock branch to remove this. I think it was necessary at the time because the default is all-or-nothing, and the condition needed to include `ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')`. However, this clause looks like it's handled by a template condition, so I think it can be completely deleted.

I'm also moving the dependsOn into the main file for better visibility.